spec/routing/single_page_routing_spec.rb

```diff
@@ -0,0 +1,35 @@
+RSpec.describe '/metrics routing' do
+  it 'routes correctly with given base path' do
+    expect(get: '/metrics/long/base/path?date_range=past-30-days').to route_to(
+      controller: 'metrics',
+      action: 'show',
+      base_path: 'long/base/path',
+      date_range: 'past-30-days'
+    )
+  end
+
+  it 'generates a route for the given base_path' do
+    expect(get: metrics_path('/base/path', date_range: 'past-30-days')).to route_to(
+      controller: 'metrics',
+      action: 'show',
+      base_path: 'base/path',
+      date_range: 'past-30-days'
+    )
+  end
```

Comment:
Whats the difference between this example and the one above?

Reply:
We are testing route generation here, the one above is testing route recognition.